app/views/spree/shared/_social.html.erb

```diff
@@ -2,10 +2,12 @@
   <% if (!spree_current_user || !spree_current_user.user_authentications) && Spree::AuthenticationMethod.active_authentication_methods? %>
     <h2><%= Spree.t(:sign_in_through_one_of_these_services) %></h2>
   <% end %>
-
-  <% Spree::AuthenticationMethod.available_for(@spree_user).each do |method| %>
-    <%= link_to(content_tag(:i, '', class: "icon-spree-#{method.provider.to_url}-circled"),
-          spree.spree_user_omniauth_authorize_url(provider: method.provider),
-          title: Spree.t(:sign_in_with, provider: method.provider)) if method.active %>
+  
+  <% if Spree::AuthenticationMethod.available_for(spree_current_user).exists? %>
```

Comment:
Adding an outer conditional doesn't change the behaviour as far as I can tell. If there aren't any, the loop never runs.